src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/InterventionService.kt

```diff
@@ -13,20 +13,27 @@ import java.util.UUID
 @Service
 class InterventionService(
   val pccRegionRepository: PCCRegionRepository,
-  val repository: InterventionRepository
+  val interventionRepository: InterventionRepository
 ) {
 
   fun getIntervention(id: UUID): InterventionDTO? {
-    return repository.findByIdOrNull(id)?.let { InterventionDTO.from(it, getPCCRegions(it)) }
+    return interventionRepository.findByIdOrNull(id)?.let { InterventionDTO.from(it, getPCCRegions(it)) }
   }
   fun getInterventionsForServiceProvider(id: AuthGroupID): List<Intervention> {
-    return repository.findByDynamicFrameworkContractServiceProviderId(id)
+    return interventionRepository.findByDynamicFrameworkContractServiceProviderId(id)
   }
   fun getAllInterventions(): List<InterventionDTO> {
-    return repository.findAll().map {
+    return interventionRepository.findAll().map {
       InterventionDTO.from(it, getPCCRegions(it))
     }
   }
+
+  fun getInterventions(pccRegionIds: List<String>): List<InterventionDTO> {
+    return interventionRepository.findByCriteria(pccRegionIds).map {
+      InterventionDTO.from(it, getPCCRegions(it))
```

Comment:
@mahalrmoj suggested we return entities from the service layer and convert to DTOs in the controller. we can do outside this PR, so just a note.